Feature/upgrade go sdk v11

ag-ramachandran opened the pull request:
**Description:** 
- Upgrade SDK for azure-kusto-go. Fixes issues for azblob updates
- Additional end to end tests. With the right env vars set, E2E tests can be performed on the exporter
- Update to Beta (stability) , since the API's have matured

**Link to tracking Issue:** 

#18886 
#18637 

**Testing:** 
![image](https://user-images.githubusercontent.com/106139410/221841830-7bbcbd7b-75b2-4421-a8b1-55adddcfbf8d.png)

End to end tests done

**Documentation:** 
- Upgrade SDK for azure-kusto-go
- Additional end to end tests
- Update to Beta (stability)

Commit:
* Rebase changes

Commit:
*Fix go imports

Commit:
* Add change logs

Commit:
* Fix yaml lint

Commit:
* Update changelog

Commit:
* Fix go mod

mx-psi commented on exporter/azuredataexplorerexporter/README.md:
```diff
@@ -2,7 +2,7 @@
 | Status                   |                       |
 | ------------------------ | --------------------- |
-| Stability                | [alpha]               |
+| Stability                | [beta]               |
```
This needs another changelog entry announcing the promotion to beta

Commit:
* Add a new change log for migrating to beta

mx-psi reviewed (commented):
There are some merge conflicts, can you fix them with a merge/rebase against main?

ag-ramachandran commented:
Sure @mx-psi, on it

Commit:
Merge branches 'main' and 'feature/UpgradeGoSDK-V11' of https://github.com/ag-ramachandran/opentelemetry-collector-contrib into feature/UpgradeGoSDK-V11

Commit:
* Update changelog in manifest

mx-psi commented:
Looks like you need to run `make gotidy`

ag-ramachandran commented:
> Looks like you need to run `make gotidy`

Acked, missed that. Running that workflow. Sorry

Commit:
*Update go mod tidy

Commit:
* Fix issue in genotelcontribcol

ag-ramachandran commented:
Hello @mx-psi , Sorry to disturb. Upgrades to this lib will need to go along with changes in azblob. How should we go about it.
```
Error: ../../receiver/azureblobreceiver/blobclient.go:30:24: undefined: azblob.ServiceClient
Error: ../../receiver/azureblobreceiver/blobclient.go:36:87: undefined: azblob.BlockBlobClient
Error: ../../receiver/azureblobreceiver/blobclient.go:66:31: undefined: azblob.NewServiceClientFromConnectionString
```

codeboten commented on .chloggen/adx-update-go-lib.yaml:
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
```
if the change is just an update to remove a deprecated library, it doesn't need a changelog entry

asaharn replied:
Thanks. Will remove it.

codeboten commented:
> Hello @mx-psi , Sorry to disturb. Upgrades to this lib will need to go along with changes in azblob. How should we go about it. [Trace](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4305280869/jobs/7507612480)
> 
> ```
> Error: ../../receiver/azureblobreceiver/blobclient.go:30:24: undefined: azblob.ServiceClient
> Error: ../../receiver/azureblobreceiver/blobclient.go:36:87: undefined: azblob.BlockBlobClient
> Error: ../../receiver/azureblobreceiver/blobclient.go:66:31: undefined: azblob.NewServiceClientFromConnectionString
> ```

I'd update both components in this pr if the dependency breaks it

asaharn committed:
Removed changelog entry

codeboten reviewed (commented):
@ag-ramachandran please resolve the conflicts

Commit:
Merge and resolve conflicts

Commit:
Merge and Resolve conflicts

Commit:
* Added tracelogs with versions for connector tracking

ag-ramachandran commented:
> @ag-ramachandran please resolve the conflicts

Hello Alex @codeboten, 
Rebased and fixed the conflicts

Failed with the following in CI though. Please let us know if there is anything we can do help on this.


`make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/oteltestbedcol'
level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package azureblobreceiver: could not load export data: no export data for \"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver\""
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package azureblobreceiver: could not load export data: no export data for \"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureblobreceiver\"\n\n"
make[2]: *** [../../Makefile.Common:166: lint] Error 3
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/configschema'
make[1]: *** [Makefile:178: cmd/configschema] Error 2
make[1]: *** Waiting for unfinished jobs....`

cc: @asaharn

ag-ramachandran committed:
Merge branch 'open-telemetry:main' into feature/UpgradeGoSDK-V11

Commit:
* Fix deprecated azure blob receiver API

ag-ramachandran commented:
Hello @codeboten , @mx-psi .Fixed receiver code for azureblobreceiver. Will be great if you can review it too

mx-psi commented on cmd/configschema/go.mod:
Where the changes on this file intentional? I think we should revert them (this is making 'checks' fail and it sounds like some command went wrong)

asaharn replied:
Hi @mx-psi , these are result go mod tidy.

mx-psi reviewed (commented):
Looks like the CI 'checks' job is failing because of changes on configschema. I think these were probably unintentional, could you take a look?

Commit:
Merge branches 'feature/UpgradeGoSDK-V11' and 'main' of https://github.com/ag-ramachandran/opentelemetry-collector-contrib into feature/UpgradeGoSDK-V11

Commit:
* Update go mod in contrib and collector

asaharn committed:
Update to go mod

asaharn committed:
Crosslink change

Commit:
* Go mod tidy changes

asaharn committed:
make crosslink changes with last tidy

asaharn committed:
make gotidy

asaharn commented:
Hi @mx-psi , ci failing due to Test failure in `translator/loki`

![image](https://user-images.githubusercontent.com/102726227/226681655-df657524-49f5-4c53-a7a7-524a3a4b0094.png)

TylerHelmuth commented:
@asaharn that test is unrelated.  This PR should take care of the issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20241

mx-psi committed:
Merge remote-tracking branch 'origin/main' into feature/UpgradeGoSDK-V11

codeboten reviewed (commented):
Needs a `make gotidy`

asaharn committed:
make gotidy changes

asaharn committed:
Merge branch 'main' of https://github.com/open-telemetry/opentelemetry-collector-contrib into feature/UpgradeGoSDK-V11

asaharn committed:
make gotidy after merge

asaharn committed:
make gotidy

asaharn committed:
make genotelcontribcol changes

asaharn commented:
HI @codeboten @mx-psi,  
Done with all the changes to make CI happy.
